src/Apps/W1/Subscription Billing/App/Usage Based Billing/Codeunits/UsageBasedContrSubscribers.Codeunit.al

```diff
@@ -269,4 +269,79 @@ codeunit 8028 "Usage Based Contr. Subscribers"
         UsageDataBilling.SetRange("Billing Line Entry No.", BillingLine."Entry No.");
         UsageDataBilling.ModifyAll("Billing Line Entry No.", BillingLineArchive."Entry No.", false);
     end;
+
+    [EventSubscriber(ObjectType::Table, Database::"Purchase Line", OnAfterDeleteEvent, '', false, false)]
+    local procedure DeleteRelatedUsageBillingLinesOnAfterDeletePurchaseLineEvent(Rec: Record "Purchase Line"; RunTrigger: Boolean)
+    var
+        UsageDataBilling: Record "Usage Data Billing";
+        PurchaseHeader: Record "Purchase Header";
+    begin
+        if not RunTrigger then
+            exit;
+        if Rec.IsTemporary then
+            exit;
+        PurchaseHeader.Get(Rec."Document Type", Rec."Document No.");
+        if not PurchaseHeader."Recurring Billing" then
+            exit;
+
+        UsageDataBilling.FilterDocumentWithLine("Service Partner"::Vendor, UsageBasedDocTypeConv.ConvertPurchaseDocTypeToUsageBasedBillingDocType(Rec."Document Type"), Rec."Document No.", Rec."Line No.");
+        if UsageDataBilling.IsEmpty() then
+            exit;
```

Comment:
There is a built in function named FilterDocumentWithLine in Usage Data Billing table. Can we use this one instead?